src/DGmethods/NumericalFluxes.jl

```diff
@@ -110,6 +110,24 @@ function numerical_boundary_flux_nondiffusive!(nf::NumericalFluxNonDiffusive,
   numerical_flux_nondiffusive!(nf, bl, F, nM, QM, auxM, QP, auxP, t)
 end
 
+"""
+    Upwind <: NumericalFluxNonDiffusive
+
+The Upwind numerical flux.
+
+# Usage
+
+    Upwind()
+
+!!! note
+
+    A balance law must provide a concrete implementation of
```

Comment:
Is there a reason the implementation of upwind flux is balance-law specific? For a given variable, the implementation should be straight forward. Is the constraint on specifying which variables get upwind fluxes?

Reply:
In general, upwinding requires solving an eigenvalue problem. You could in theory for linear systems, we could specify the flux as `F = A * q` for some matrix `A` and then use an eigenvalue solver to diagonalize etc, but since this would need to be done at every degree of freedom it would likely not be efficient.

For nonlinear problems and problems with discontinuous coefficients this is even more challenging.

Reply:
Note also that there isn't just one variable that gets upwinded, but its a combination of the variables because it comes from the left and right eigenvectors, and looking at the signs of the eigenvalues.